### PR TITLE
Update split_data with new default for timeseries

### PR DESCRIPTION
**Pull request by eccabay**
Reduces the default `test_size` in `split_data` to be 0.1 instead of 0.2, as long as the new value is greater than the passed in forecast horizon.

### CODE REVIEW

**Review by fjlanasa — APPROVED** [review]

File: evalml/tests/preprocessing_tests/test_split_data.py [hunk 1]
```diff
@@ -59,3 +59,97 @@ def test_split_data(
...
+    if is_multiclass(problem_type):
+        X, y = X_y_multi
+    if is_regression(problem_type):
+        X, y = X_y_regression
+    problem_configuration = None
+    if is_time_series(problem_type):
+        problem_configuration = {"gap": 1, "max_delay": 7, "time_index": "ts_data"}
+
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
```
Comment (chukarsten) [inline evalml/tests/preprocessing_tests/test_split_data.py:85]: I suspect this is boiler plate not just for these two tests but also across many of the tests.  Did you want to refactor it into something that returns `problem_configuration`, `X`, and `y` given `data_type`, `problem_type`, etc?  I am pretty sure I've seen this a few other places too.  I think it's also fine to file an issue to call out the refactor with the modules we should refactor this code into.

File: evalml/tests/preprocessing_tests/test_split_data.py [hunk 2]
```diff
@@ -59,3 +59,97 @@ def test_split_data(
...
+        y,
+        problem_type="time series regression",
+        problem_configuration=problem_configuration,
+    )
+    test_size = fh
+    train_size = len(X) - fh
+    assert len(X_train) == train_size
+    assert len(X_test) == test_size
+    assert len(y_train) == train_size
+    assert len(y_test) == test_size
```
Comment (chukarsten) [inline evalml/tests/preprocessing_tests/test_split_data.py:155]:
```suggestion
@pytest.mark.parametrize("test", ["fh_limitation", "no_fh_limitation"])
def test_split_data_ts(X_y_regression, test):
    X, y = X_y_regression

    if test == "no_fh_limitation":
        test_pct = 0.1
        fh = 5
        test_size = len(X) * test_pct
        train_size = len(X) - test_size
	elif test == "fh_limitation":
		fh = 25    
		test_size = fh
        train_size = len(X) - fh
    problem_configuration = {
        "gap": 1,
        "max_delay": 7,
        "forecast_horizon": fh,
        "time_index": "ts_data",
    }
    X_train, X_test, y_train, y_test = split_data(
        X,
        y,
        problem_type="time series regression",
        problem_configuration=problem_configuration,
    )
    assert len(X_train) == train_size
    assert len(X_test) == test_size
    assert len(y_train) == train_size
    assert len(y_test) == test_size
```

Comment (chukarsten) [reply evalml/tests/preprocessing_tests/test_split_data.py:155]: I'm a little bigger of a fan using parametrization to drop in a nice string that replaces your in-line comments, so when the test case fails, it leaves a nice message.  It also allows both test cases to run if the first half test case fails!  I am not sure the suggestion runs though, my IDE was acting weird.

**Review by chukarsten — APPROVED** [review]
LGTM, just added a few suggestions, take them or leave them.  Nothing blocking!

File: evalml/tests/preprocessing_tests/test_split_data.py (hunk 1 above)
Comment (eccabay) [reply evalml/tests/preprocessing_tests/test_split_data.py:85]: I like this. A slight refactor to `get_test_data_from_configuration` works wonderfully to factor this out! We can go through and replace other instances in a future story.

**Pull request merged**